test/unit/player_test.rb

```diff
@@ -0,0 +1,55 @@
+require 'test_helper'
+
+class PlayerTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
+
+  test "validates for name, level, armor class, and passive perception" do
+    player = Player.new
+    assert !player.save
+    player.name = "bob"
+    assert !player.save
+    player.level = 3
+    assert !player.save
+    player.armor_class = 14
+    assert !player.save
+    player.passive_perception = 3
+    assert player.save
+  end
+
+  test "set health and initiative" do
+    player = players(:valid_player)
+    player.battle_setup(5, 23)
+    assert player.initiative == 5
+    assert player.current_health == 23
+  end
+
+  test "change health adds if number is positive" do
+    player = players(:valid_player)
+    player.battle_setup(5, 20)
+    player.change_health(5)
+    assert player.current_health == 25
+  end
+
+  test "change health subtracts if number is negative" do
+    player = players(:valid_player)
+    player.battle_setup(2, 20)
+    player.change_health(-5)
+    assert player.current_health == 15
+  end
+
+  test "level up adds 1 to level" do
+    player = players(:valid_player)
+    player_level = player.level
+    player.level_up
+    assert player.level == (player_level + 1)
```

Comment:
should use assert_equal instead of assert <blah>, assert can be misleading and will be true for any case that the thing it is passed is valid-- which may not be what we're testing for. Using assert_equal will also make your tests easier to read, and your output easier to interpret. (It'll say "Expected <BLAH>, Received: <STUFF>")